math/sieve_of_eratosthenes.cpp

```diff
@@ -49,17 +43,22 @@ void print(uint32_t N) {
 /**
  * Initialize the array
  */
-void init() {
-    for (uint32_t i = 1; i < MAX; i++) {
+void init(uint32_t N, bool *isprime) {
+    for (uint32_t i = 0; i < N; i++) {
         isprime[i] = true;
     }
 }
 
-/** main function */
+/**
+ * Main function
+ */
 int main() {
     uint32_t N = 100;
-    init();
-    sieve(N);
-    print(N);
+    bool *isprime = new bool[N];
+    init(N, isprime);
```

Comment:
```suggestion
    memset(isprime, true, N * sizeof(bool)); // preset all numbers to `true`
```
this on-line is same as what the `init` function does above. Hence, if you use this line, you can delete the init function.
You might have to include `<cstring>` to use this function.